test/basics.jl

```diff
@@ -1,57 +1,41 @@
-test_string_length = 100
+test_string_length = 1:256
 
-#string_types = [ASCIIStr, LatinStr, UTF8Str, UCS2Str, UTF16Str, UTF32Str, UniStr]
-string_types = [ASCIIStr, LatinStr, UTF8Str, UTF16Str, UTF32Str]
+AllCharTypes = (ASCIIChr, LatinChr, UCS2Chr, UTF32Chr);
+compat_types = Dict(ASCIIStr => (ASCIIChr, ), LatinStr => (ASCIIChr, LatinChr), UCS2Str => (ASCIIChr, LatinChr, UCS2Chr))
+for T in (String, UTF8Str, UTF16Str, UTF32Str)
+    compat_types[T] = AllCharTypes
+end
+string_types = keys(compat_types)
 
 ##  create type specific test strings
-test_strings_base = Dict{String, Any}()
-for T in string_types
-    test_strings_base["$T"] = join([randchar(Strs.codepoint_type(T)) for i in 1:test_string_length])
+test_strings_base = Dict()
+for T in AllCharTypes
+    clist = String[]
```

Comment:
`test_strings_base[T] = [String([convert(Char, randchar(T)%UInt32) for i = 1:len]) for len in test_string_length]`